### ISSUE TITLE
Version 2.5: The "path" argument must be of type string. Received type undefined

### ISSUE THREAD

**Issue by Puritanic** [rokucommunity/vscode-brightscript-language#259]
The extension started crashing with the new version.

Notification errors:

```
Critical error parsing/ validating /Users/dtasevski/Redbox/roku/src/components/BaseScene/BaseScene.brs: Cannot read property 'hasFile' of undefined
```

```
BrighterScript language server failed to start: 
The "path" argument must be of type string. Received type undefined
```

The output from the Brighterscript Language Server:

```
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type undefined
    at validateString (internal/validators.js:112:11)
    at Object.resolve (path.js:981:7)
    at RokuDeploy.getOptions (/Users/dtasevski/.vscode-insiders/extensions/celsoaf.brightscript-2.5.0/node_modules/brighterscript/node_modules/roku-deploy/dist/RokuDeploy.js:985:51)
    at Util.normalizeConfig (/Users/dtasevski/.vscode-insiders/extensions/celsoaf.brightscript-2.5.0/node_modules/brighterscript/dist/util.js:237:88)
    at Util.normalizeAndResolveConfig (/Users/dtasevski/.vscode-insiders/extensions/celsoaf.brightscript-2.5.0/node_modules/brighterscript/dist/util.js:211:27)
    at ProgramBuilder.run (/Users/dtasevski/.vscode-insiders/extensions/celsoaf.brightscript-2.5.0/node_modules/brighterscript/dist/ProgramBuilder.js:70:46)
    at LanguageServer.createWorkspace (/Users/dtasevski/.vscode-insiders/extensions/celsoaf.brightscript-2.5.0/node_modules/brighterscript/dist/LanguageServer.js:290:39)
    at async Promise.all (index 0)
    at async LanguageServer.onInitialized (/Users/dtasevski/.vscode-insiders/extensions/celsoaf.brightscript-2.5.0/node_modules/brighterscript/dist/LanguageServer.js:154:13)
(node:8786) UnhandledPromiseRejectionWarning: TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type undefined
    at validateString (internal/validators.js:112:11)
    at Object.isAbsolute (path.js:1030:5)
    at RokuDeploy.getDestPath (/Users/dtasevski/.vscode-insiders/extensions/celsoaf.brightscript-2.5.0/node_modules/brighterscript/node_modules/roku-deploy/dist/RokuDeploy.js:376:18)
    at LanguageServer.synchronizeStandaloneWorkspaces (/Users/dtasevski/.vscode-insiders/extensions/celsoaf.brightscript-2.5.0/node_modules/brighterscript/dist/LanguageServer.js:511:39)
    at LanguageServer.reloadWorkspaces (/Users/dtasevski/.vscode-insiders/extensions/celsoaf.brightscript-2.5.0/node_modules/brighterscript/dist/LanguageServer.js:466:20)
    at async LanguageServer.onDidChangeWatchedFiles (/Users/dtasevski/.vscode-insiders/extensions/celsoaf.brightscript-2.5.0/node_modules/brighterscript/dist/LanguageServer.js:563:13)
(node:8786) UnhandledPromiseRejectionWarning: TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type undefined
    at validateString (internal/validators.js:112:11)
    at Object.isAbsolute (path.js:1030:5)
    at RokuDeploy.getDestPath (/Users/dtasevski/.vscode-insiders/extensions/celsoaf.brightscript-2.5.0/node_modules/brighterscript/node_modules/roku-deploy/dist/RokuDeploy.js:376:18)
    at LanguageServer.synchronizeStandaloneWorkspaces (/Users/dtasevski/.vscode-insiders/extensions/celsoaf.brightscript-2.5.0/node_modules/brighterscript/dist/LanguageServer.js:511:39)
    at LanguageServer.reloadWorkspaces (/Users/dtasevski/.vscode-insiders/extensions/celsoaf.brightscript-2.5.0/node_modules/brighterscript/dist/LanguageServer.js:466:20)
    at async LanguageServer.onDidChangeWatchedFiles (/Users/dtasevski/.vscode-insiders/extensions/celsoaf.brightscript-2.5.0/node_modules/brighterscript/dist/LanguageServer.js:563:13)
(node:8786) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 4)
(node:8786) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 4)
(node:8786) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
(node:8786) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
(node:8786) UnhandledPromiseRejectionWarning: TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type undefined
    at validateString (internal/validators.js:112:11)
    at Object.isAbsolute (path.js:1030:5)
    at RokuDeploy.getDestPath (/Users/dtasevski/.vscode-insiders/extensions/celsoaf.brightscript-2.5.0/node_modules/brighterscript/node_modules/roku-deploy/dist/RokuDeploy.js:376:18)
    at LanguageServer.synchronizeStandaloneWorkspaces (/Users/dtasevski/.vscode-insiders/extensions/celsoaf.brightscript-2.5.0/node_modules/brighterscript/dist/LanguageServer.js:511:39)
    at LanguageServer.reloadWorkspaces (/Users/dtasevski/.vscode-insiders/extensions/celsoaf.brightscript-2.5.0/node_modules/brighterscript/dist/LanguageServer.js:466:20)
    at async LanguageServer.onDidChangeWatchedFiles (/Users/dtasevski/.vscode-insiders/extensions/celsoaf.brightscript-2.5.0/node_modules/brighterscript/dist/LanguageServer.js:563:13)
(node:8786) UnhandledPromiseRejectionWarning: TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type undefined
    at validateString (internal/validators.js:112:11)
    at Object.isAbsolute (path.js:1030:5)
    at RokuDeploy.getDestPath (/Users/dtasevski/.vscode-insiders/extensions/celsoaf.brightscript-2.5.0/node_modules/brighterscript/node_modules/roku-deploy/dist/RokuDeploy.js:376:18)
    at LanguageServer.synchronizeStandaloneWorkspaces (/Users/dtasevski/.vscode-insiders/extensions/celsoaf.brightscript-2.5.0/node_modules/brighterscript/dist/LanguageServer.js:511:39)
    at LanguageServer.reloadWorkspaces (/Users/dtasevski/.vscode-insiders/extensions/celsoaf.brightscript-2.5.0/node_modules/brighterscript/dist/LanguageServer.js:466:20)
    at async LanguageServer.onDidChangeWatchedFiles (/Users/dtasevski/.vscode-insiders/extensions/celsoaf.brightscript-2.5.0/node_modules/brighterscript/dist/LanguageServer.js:563:13)
(node:8786) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 6)
(node:8786) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 6)
(node:8786) UnhandledPromiseRejectionWarning: TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type undefined
    at validateString (internal/validators.js:112:11)
    at Object.isAbsolute (path.js:1030:5)
    at RokuDeploy.getDestPath (/Users/dtasevski/.vscode-insiders/extensions/celsoaf.brightscript-2.5.0/node_modules/brighterscript/node_modules/roku-deploy/dist/RokuDeploy.js:376:18)
    at LanguageServer.synchronizeStandaloneWorkspaces (/Users/dtasevski/.vscode-insiders/extensions/celsoaf.brightscript-2.5.0/node_modules/brighterscript/dist/LanguageServer.js:511:39)
    at LanguageServer.reloadWorkspaces (/Users/dtasevski/.vscode-insiders/extensions/celsoaf.brightscript-2.5.0/node_modules/brighterscript/dist/LanguageServer.js:466:20)
    at async LanguageServer.onDidChangeWatchedFiles (/Users/dtasevski/.vscode-insiders/extensions/celsoaf.brightscript-2.5.0/node_modules/brighterscript/dist/LanguageServer.js:563:13)
(node:8786) UnhandledPromiseRejectionWarning: TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type undefined
    at validateString (internal/validators.js:112:11)
    at Object.isAbsolute (path.js:1030:5)
    at RokuDeploy.getDestPath (/Users/dtasevski/.vscode-insiders/extensions/celsoaf.brightscript-2.5.0/node_modules/brighterscript/node_modules/roku-deploy/dist/RokuDeploy.js:376:18)
    at LanguageServer.synchronizeStandaloneWorkspaces (/Users/dtasevski/.vscode-insiders/extensions/celsoaf.brightscript-2.5.0/node_modules/brighterscript/dist/LanguageServer.js:511:39)
    at LanguageServer.reloadWorkspaces (/Users/dtasevski/.vscode-insiders/extensions/celsoaf.brightscript-2.5.0/node_modules/brighterscript/dist/LanguageServer.js:466:20)
    at async LanguageServer.onDidChangeWatchedFiles (/Users/dtasevski/.vscode-insiders/extensions/celsoaf.brightscript-2.5.0/node_modules/brighterscript/dist/LanguageServer.js:563:13)
(node:8786) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 8)
(node:8786) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 8)
```

VSCode version and OS info:
```
Version: 1.47.0-insider
Commit: 59c7cbcf15745e4a8adb4fc59b63aa93f14d3cc3
Date: 2020-07-08T05:17:07.920Z (3 hrs ago)
Electron: 7.3.2
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Darwin x64 19.5.0
```

**Comment by TwitchBronBron** [comment]
Does the extension crash every time it starts, or is this just occasionally?

**Comment by Puritanic** [comment]
It's crashing every time I open the project, and the error notifications are appearing each time I save brightscript or XML file.

**Comment by Puritanic** [comment]
I think that I've found out the culprit for extension crashing. Here is the project's `bsconfig` file content:
```
{
	"rootDir": "./src",
	"watch": false,
	"stagingFolderPath": "./out/temp/",
	"files": [
		"components/**/*.*",
		"images/**/*.*",
		"source/**/*.*",
		"manifest",
		"queries/*",
		"fonts/*",
		"lib/*"
	],
	"diagnosticFilters": [
		{
			// Disable "overrides function" rule
			"codes": [1010, 1013]
		}
	]
}
```
Removing `"stagingFolderPath": "./out/temp/"` from it solves the crashing problem, but I now can't build a zip file with `bsc`, possibly related to this issue in brighterscript repo: https://github.com/rokucommunity/brighterscript/issues/127.

**Comment by TwitchBronBron** [comment]
This should be fixed in [v2.5.4](https://github.com/rokucommunity/vscode-brightscript-language/releases/tag/v2.5.4). Feel free to re-open if there are further issues.
